opensearch-api/lib/opensearch/api/actions/cat/pit_segments.rb

```diff
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+module OpenSearch
+  module API
+    module Cat
+      module Actions
+        # Retrieves info of certain PIT segments
+        #
+        # @option arguments [Hash] body: Must include `pit_id`, which is an array of PIT IDs. (required)
+        # @option arguments [String] :format a short version of the Accept header, e.g. json, yaml
+        # @option arguments [List] :h Comma-separated list of column names to display
+        # @option arguments [Boolean] :help Return help information
+        # @option arguments [List] :s Comma-separated list of column names or column aliases to sort by
+        # @option arguments [Boolean] :v Verbose mode. Display column headers
+        # @option arguments [Hash] :headers Custom HTTP headers
+        def pit_segments(arguments = {})
+          raise ArgumentError, "Required argument 'body' missing" unless arguments[:body]
+
+          headers = arguments.delete(:headers) || {}
```

Comment:
Not sure if this is a pattern elsewhere, same as above, don't modify incoming hashes.

Reply:
It's a pattern seen in other API methods. Right below this line you will see `arguments = arguments.clone`, which is unusual considering that it could have been the first line of the method. The future generated methods will "fix" these issues. I just applied the same pattern for consistency's sake but performing a `deep_dup` before modifying any inputs for these new APIs works too.

Reply:
I opened https://github.com/opensearch-project/opensearch-ruby/issues/158. Here are some stories about why: https://code.dblock.org/2020/11/11/dup-dont-modify-input-hash-parameters-in-ruby.html